### PR TITLE
initial integration

### PR DESCRIPTION
**Pull request by vigan-abd**

### CODE REVIEW

File: src/promise-timeout.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+const promiseTimeout = async (promise, ms, errMsg = null) => {
+  const _promise = typeof promise === 'function' ? promise() : promise
+  let timeout = null
+
+  const res = await Promise.race([
+    _promise,
+    new Promise(resolve => {
+      timeout = setTimeout(() => {
+        resolve(new PromiseTimeoutError(errMsg))
```
Comment (robertkowalski) [inline src/promise-timeout.js:34]: shouldn't it be reject?

File: src/promise-timeout.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      }, ms)
+    })
+  ])
+
+  if (res instanceof PromiseTimeoutError) throw res
```
Comment (robertkowalski) [inline src/promise-timeout.js:39]: what happens with other exceptions? are they "eaten up"? can you add a test for the case?

**Review by robertkowalski — COMMENTED** [review]
one more test would be great!

File: src/promise-timeout.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * Run a promise with an execution timeout
+ *
+ * @param {Promise<any>|Function} promise Promise that timeout will be checked against
+ * @param {number} ms Time in milliseconds until promise is timed out
+ * @param {string} [errMsg] Custom error message for PromiseTimeoutError
+ *
+ * @returns {Promise<any>}
+ * @throws {PromiseTimeoutError}
+ */
+const promiseTimeout = async (promise, ms, errMsg = null) => {
```
Comment (robertkowalski) [inline src/promise-timeout.js:26]: check this:

```js
diff --git a/src/promise-timeout.js b/src/promise-timeout.js
index 3c3fd06..a57b1d2 100644
--- a/src/promise-timeout.js
+++ b/src/promise-timeout.js
@@ -26,12 +26,13 @@ class PromiseTimeoutError extends Error {
 const promiseTimeout = async (promise, ms, errMsg = null) => {
   const _promise = typeof promise === 'function' ? promise() : promise
   let timeout = null
+  const err = new PromiseTimeoutError(errMsg)

   const res = await Promise.race([
     _promise,
     new Promise(resolve => {
       timeout = setTimeout(() => {
-        resolve(new PromiseTimeoutError(errMsg))
+        resolve(err)
       }, ms)
     })
   ])
```

then check the stacktrace of the error :)

File: src/promise-timeout.js (hunk 2 above)
Comment (vigan-abd) [reply src/promise-timeout.js:39]: yep we already have test case for other errors: https://github.com/bitfinexcom/lib-js-util-promise/pull/1/files#diff-5cf8949559fe57d1e5466ca328999b67cc3f6e2ddc99d38fc3ec3c2454edc753R51

File: src/promise-timeout.js (hunk 1 above)
Comment (vigan-abd) [reply src/promise-timeout.js:34]: could be rewritten also as reject, but preferred to resolve it so only promise param could throw an exception/rejection. If you think that rejection would be more convenient then I'll wrap it with a try catch and filter err

File: src/promise-timeout.js (hunk 2 above)
Comment (vigan-abd) [reply src/promise-timeout.js:39]: hmm actually agree, looks more convenient with rejection

**Review by bariscicek — APPROVED** [review]
LGTM!

**Review by robertkowalski — APPROVED** [review]
lgtm

**Pull request merged**